### PR TITLE
Disabled and fixed eslint rules; bug fixes

### PR DESCRIPTION
**Pull request by SADmammoth**
Disabled rule for all files:
* react/jsx-props-no-spreading

Disabled rules for som lines/files:
* react/jsx-one-expression-per-line
* react/require-default-props
* react/button-has-type
* react/no-unused-prop-types
* react/sort-comp
* jsx-a11y/no-onchange
* jsx-a11y/click-events-have-key-events
* jsx-a11y/no-noninteractive-element-interactions
* no-param-reassign
* camelcase

### CODE REVIEW

File: src/components/elements/DialogButton/DialogButton.js [hunk 1]
```diff
@@ -25,8 +33,10 @@ const DialogButton = (props) => {
     setLoading(true);
 
     return propsOnSubmit(data).then((response) => {
-      setLoading(false);
-      reload();
+      if (!deletesAfterConfirmation) {
```
Comment (maximsan) [inline src/components/elements/DialogButton/DialogButton.js:36]: maybe you wanted to name it `deleteAfterConfirmation` ?

File: src/components/elements/DialogButton/DialogButton.js [hunk 2]
```diff
@@ -7,7 +7,15 @@ import ButtonGroup from '../ButtonGroup';
...
+    buttonClassMod,
+    dialogValue,
+    message,
+    confirmButtonClassMod,
+    confirmButtonContent,
+    deletesAfterConfirmation,
+  } = props;
 
   const [loading, setLoading] = useState(false);
   const modal = useRef({});
```
Comment (maximsan) [inline src/components/elements/DialogButton/DialogButton.js:21]: it's better set to null

File: src/helpers/Client/ClientFirebase.js [hunk 3]
```diff
@@ -199,7 +210,8 @@ const Client = {
       return { userId: doc.data().userID, memberTaskId: doc.id };
     };
 
-    return Promise.allSettled(membersTasks.docs.map(formatAssigned));
+    const assigned = (await Promise.allSettled(membersTasks.docs.map(formatAssigned))).map(getValue);
```
Comment (maximsan) [inline src/helpers/Client/ClientFirebase.js:213]: split the code in several operations

File: src/helpers/Client/ClientFirebase.js [hunk 4]
```diff
@@ -211,27 +223,32 @@ const Client = {
       assigned[taskId] = assignedArray;
     };
 
-    await Promise.allSettled(taskIds.map(processAssigned));
+    (await Promise.allSettled(taskIds.map(processAssigned))).map(getValue);
```
Comment (maximsan) [inline src/helpers/Client/ClientFirebase.js:226]: split the code in several operations

File: src/helpers/Client/getValue.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export default function getValue(response) {
+  return response.value;
```
Comment (maximsan) [inline src/helpers/Client/getValue.js:2]: does it make sense ? you have spread operator to get just `value`...

File: src/components/elements/DialogButton/DialogButton.js (hunk 2 above)
Comment (SADmammoth) [reply src/components/elements/DialogButton/DialogButton.js:21]: fixed

File: src/helpers/Client/ClientFirebase.js (hunk 4 above)
Comment (SADmammoth) [reply src/helpers/Client/ClientFirebase.js:226]: fixed

File: src/helpers/Client/ClientFirebase.js (hunk 3 above)
Comment (SADmammoth) [reply src/helpers/Client/ClientFirebase.js:213]: fixed

**Pull request merged**